lib/galaxy/workflow/modules.py

```diff
@@ -1600,6 +1651,24 @@ def step_state_to_tool_state(self, state):
         }
         if multiple is not None:
             state["parameter_definition"]["multiple"] = multiple
+        if source_validators is not None:
+            form_validators = []
+            # the form definition can change from Galaxy to Galaxy fairly freely, but the source validators are persisted
+            # and need to be consistent - here we convert the persisted/YAML tool definition version to the "tool form" version.
+            for i, source_validator in enumerate(source_validators):
+                source_type = source_validator["type"]
+                if source_type == "regex":
+                    form_validators.append(
+                        {
+                            "__index__": i,
```

Comment:
Why do we we need the `__index__` field ? I thought this was legacy stuff from way back ?

Reply:
Just felt safer since it is only used by the client UI. I tried dropping it though and everything seems fine still.